### PR TITLE
issue: 844121 Fixed bug in the stats epoll and split segment added

### PR DESCRIPTION
**Pull request by serglelv**
Fixed bug in the stats remove epoll block. Added split the segment in case of a small window. Please review.
@igor-ivanov @OphirMunk @NirNitzani @alexvm

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/483/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/498/ for details (Mellanox internal link).

**Comment by rosenbaumalex** [comment]
Sergey, 

1. Aren't these cheery-picked from master? Would be good to specify that and reference sha in commit message. 

2. Better to push on two seperate commits to keep things clear. 

Alex

File: src/stats/stats_publisher.cpp [hunk 1]
```diff
@@ -604,8 +604,24 @@ void vma_stats_instance_create_epoll_block(int fd, iomux_func_stats_t* local_sta
...
-                ep_stats->enabled = false;
+	iomux_func_stats_t* ep_func_stats = (iomux_func_stats_t*)g_p_stats_data_reader->pop_p_skt_stats(local_stats_addr);
+
+	if (NULL == ep_func_stats) {
+		vlog_printf(VLOG_DEBUG, "%s:%d: application vma_stats pointer is NULL\n", __func__, __LINE__);
+		g_lock_ep_stats.unlock();
+		return;
+	}
+	// Search ep_mem block to release
+	for (int i=0; i<NUM_OF_SUPPORTED_EPFDS; i++) {
```
Comment (igor-ivanov) [inline src/stats/stats_publisher.cpp]: @serglelv consider meeting to coding style as `for (int i = 0; i < NUM_OF_SUPPORTED_EPFDS; i++) {`

**Comment by igor-ivanov** [comment]
@serglelv it looks as issue not specific for experimental_256B branch. I would like to suggest creating new issue for master firstly and push PR to master too.

**Comment by OphirMunk** [comment]
This PR is a cherry-pick of 2 PRs from master branch:
d76177e issue: 784234 fixed epoll is not cleared from vma_stats [Daniel Libenson]
d699f25 issue: 802922 LWIP - Split head of TCP unsent segment when bigger than send window size and NODELAY flag is set [Daniel Libenson] 

@serglelv please indicate if this fix is based on bugs you found in tests

If not - I will close this PR as merging experimental_256B and master branches is a parallel task

**Comment by serglelv** [comment]
2 PRs are needed but the root casuse related to the race condition and the fix is WIP.   
@OphirMunk

**Comment by OphirMunk** [comment]
Closing this PR. Fixes will be part of merge from master branch. 
If this PR is still needed in experimental branch - can reopen it.

**Pull request closed without merging**